extensions/hibernate-validator/deployment/src/test/java/io/quarkus/hibernate/validator/test/RepeatedConstraintsTest.java

```diff
@@ -0,0 +1,98 @@
+package io.quarkus.hibernate.validator.test;
+
+import static java.lang.annotation.ElementType.ANNOTATION_TYPE;
+import static java.lang.annotation.ElementType.CONSTRUCTOR;
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.ElementType.TYPE_USE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.validation.Constraint;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import javax.validation.Payload;
+import javax.validation.ValidatorFactory;
+import javax.validation.constraints.Pattern;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.hibernate.validator.test.RepeatedConstraintsTest.MyConstraint.List;
+import io.quarkus.test.QuarkusUnitTest;
+
+public class RepeatedConstraintsTest {
+
+    @Inject
+    ValidatorFactory validatorFactory;
+
+    @RegisterExtension
+    static final QuarkusUnitTest test = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class).addClasses(TestBean.class));
+
+    @Test
+    public void testConstraints() {
+        assertThat(validatorFactory.getValidator().validate(new TestBean("INVALID", "INVALID")))
+                .hasSize(4);
+    }
+
+    static class TestBean {
+
+        public TestBean(String testValue1, String testValue2) {
+            super();
+            this.testValue1 = testValue1;
+            this.testValue2 = testValue2;
+        }
+
+        @Pattern(regexp = ".*[0-9]+.*", message = "Error message")
+        @Pattern(regexp = ".*[a-z]+.*", message = "Error message")
+        public String testValue1;
+
+        @MyConstraint("value1")
+        @MyConstraint("value2")
+        public String testValue2;
```

Comment:
Can you separate the tests in 2 bean classes?

Because, if at least one works, then all the constraints will be detected so we really need to test the built-in constraints and the custom constraints in 2 separate beans.

Reply:
ok sure